pkg/middlewares/auth/forward.go

```diff
@@ -158,7 +160,22 @@ func (fa *forwardAuth) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	fa.next.ServeHTTP(rw, req)
 }
 
-func writeHeader(req, forwardReq *http.Request, trustForwardHeader bool) {
+func filterForwardRequestHeaders(forwardRequestHeaders http.Header, allowedHeaders []string) http.Header {
+	if len(allowedHeaders) == 0 {
+		return forwardRequestHeaders
+	}
+
+	filteredHeaders := http.Header{}
+	for _, headerName := range allowedHeaders {
+		headerKey := http.CanonicalHeaderKey(headerName)
+		if len(forwardRequestHeaders[headerKey]) > 0 {
+			filteredHeaders[headerKey] = append([]string(nil), forwardRequestHeaders[headerKey]...)
+		}
+	}
+	return filteredHeaders
+}
+
+func writeHeader(req, forwardReq *http.Request, trustForwardHeader bool, allowedHeaders []string) {
 	utils.CopyHeaders(forwardReq.Header, req.Header)
```

Comment:
I would apply the filtering before the `X-Forwarded-` headers handling.
It would guarantee their presence and allow the new `authRequestHeaders` option to not interfere with the `trustForwardHeader' option.
WDYT ?
(Unfortunately that imply to update your tests and also the documentation to use other examples)

```suggestion
	utils.CopyHeaders(forwardReq.Header, req.Header)
	forwardReq.Header = filterForwardRequestHeaders(forwardReq.Header, allowedHeaders)
```

Reply:
I also was thinking what is best place for. Thank you for a good catch about trustForwardHeader option interference. I foing to do it.

The update tests for this particular case not a problem.